dpx/src/dpx_pdfnames.rs

```diff
@@ -263,30 +262,20 @@ pub unsafe extern "C" fn pdf_names_close_object(
     0i32
 }
 #[inline]
-unsafe extern "C" fn cmp_key(mut d1: *const libc::c_void, mut d2: *const libc::c_void) -> i32 {
-    let mut cmp;
-    let sd1 = d1 as *const named_object;
-    let sd2 = d2 as *const named_object;
-    if (*sd1).key.is_null() {
-        cmp = -1i32
-    } else if (*sd2).key.is_null() {
-        cmp = 1i32
+fn cmp_key(sd1: &named_object, sd2: &named_object) -> ::core::cmp::Ordering {
+    use ::core::cmp::Ordering;
+    use ::core::slice;
+    if sd1.key.is_null() {
+        Ordering::Less
+    } else if sd2.key.is_null() {
+        Ordering::Greater
     } else {
-        let keylen = if (*sd1).keylen < (*sd2).keylen {
-            (*sd1).keylen
-        } else {
-            (*sd2).keylen
-        };
-        cmp = memcmp(
-            (*sd1).key as *const libc::c_void,
-            (*sd2).key as *const libc::c_void,
-            keylen as _,
-        );
-        if cmp == 0 {
-            cmp = (*sd1).keylen - (*sd2).keylen
+        unsafe {
+            let key1 = slice::from_raw_parts(sd1.key, sd1.keylen as usize);
```

Comment:
I sense that you could *almost* turn key/keylen into a String. Something for another PR, but this seems relatively self-contained at first glance, you just need to get it back to raw parts when it’s consumed by `pdf_new_string`. (Which we want to be String-based with lopdf eventually.) Is that feasible do you think?

Reply:
Good idea, I'll try it.

Reply:
(Maybe has to be a CString for now.)